server/server.js

```diff
@@ -140,25 +155,27 @@ function movePeriodic(ctrlls) {
       y = y + height;
     }
     ctrlls[key].y = y;
+    ctrl.angle = angle;
+    ctrl.speed = speed;
   }
 }
 
-
 function moveBounded(ctrlls) {
-  for (let key in ctrlls) {
-    let x = ctrlls[key].x;
-    let y = height - ctrlls[key].y;
-    const angle = Math.PI/2 - ctrlls[key].angle;
-    const speed = ctrlls[key].speed*5/fps;
-    // console.log(x,y, rad, key)
-    x += Math.ceil(speed*Math.cos(angle));
-    y += Math.ceil(speed*Math.sin(angle));
-    // console.log(x, y);
-    if (x < width - padding && x > padding) {
-      ctrlls[key].x = x;
-    }
-    if (y < height - padding && y > padding) {
-      ctrlls[key].y = height - y;
+  const keys = Object.keys(ctrlls);
+  for (let i=0; i<keys.length ; i++) {
```

Comment:
Do you think that eventually `movePeriodic` and `moveBounded` could be merged? They do fundamentally similar things, they duplicate a lot of code, but sometimes slightly differently (such as iterating the controllers here).